File: Domain/Recipe.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Recipe
    {
        public Guid Id { get; set; }

        public string Title { get; set; }

        public string Description { get; set; }

        public string Source { get; set; }

        public string Url { get; set; }

        public string Notes { get; set; }

        public string PrepTime { get; set; }

        public string CookTime { get; set; }

        public bool IsPrivate { get; set; }

        public virtual ICollection<UserRecipe> UserRecipes { get; set; }
    }
}

File: client-app/src/app/models/recipe.ts
export interface IRecipe {
  id: string;
  title: string;
  description: string;
  source: string;
  url: string;
  notes: string;
  prepTime: string;
  cookTime: string;
  isPrivate: boolean;
  isFollowing: boolean;
  isCreator: boolean;
  followers: IFollower[];
}

export interface IRecipeFormValues extends Partial<IRecipe> {}

export class RecipeFormValues implements IRecipeFormValues {
  id?: string = undefined;
  title: string = '';
  description: string = '';
  source: string = '';
  url: string = '';
  notes: string = '';
  prepTime: string = '';
  cookTime: string = '';
  isPrivate: boolean = false;

  /**
   *
   */
  constructor(init?: IRecipeFormValues) {
    Object.assign(this, init);
  }
}

export interface IFollower {
  username: string;
  displayName: string;
  image: string;
  isCreator: boolean;
}
